pkg/apis/pipeline/v1alpha1/taskrun_validation.go

```diff
@@ -69,6 +70,18 @@ func (ts *TaskRunSpec) Validate() *apis.FieldError {
 			return err
 		}
 	}
+
+	if ts.PipelineRunTimeout != nil {
+		// timeout should be a valid duration of between 0 and 24 hours.
+		maxTimeout := time.Duration(24 * time.Hour)
+
+		if ts.PipelineRunTimeout.Time.After(time.Now().Add(maxTimeout)) {
+			return apis.ErrInvalidValue(fmt.Sprintf("Time from now until timeout %s should be < 24h", ts.PipelineRunTimeout.Sub(time.Now()).Truncate(time.Second).String()), "spec.pipelineruntimeout")
+		} else if ts.PipelineRunTimeout.Time.Before(time.Now()) {
+			return apis.ErrInvalidValue(fmt.Sprintf("Time from now until timeout %s should be >= 0", ts.PipelineRunTimeout.Sub(time.Now()).Truncate(time.Second).String()), "spec.pipelineruntimeout")
+		}
+	}
```

Comment:
could this logic be shared with the pipeline logic? e.g. moved into a module they can both call (i think it's the same)

Reply:
So after removing `PipelineRunTimeout`, this goes away too, so hey. =)